### PR TITLE
added support for async serialize

### PR DESCRIPTION
**Pull request by pon**
## What

Adds support for the `serialize` method to return a promise

## Why

Async serialization FTW

@mgartner @elnaz @robinjoseph08 @brianseitel

### CODE REVIEW

**Comment by mgartner** [comment]
LGTM

File: test/index.test.js [hunk 1]
```diff
@@ -80,6 +106,51 @@ describe('serializer plugin', function () {
       });
     });
 
+    it('should return formatted collection when serialize uses promise', function (done) {
```
Comment (robinjoseph08) [inline test/index.test.js]: should there be another test checking the an array of PromiseModels as opposed to a collection? there's a test for an array of ErrorModels.

Comment (pon) [inline test/index.test.js]: Yeah I can add that for completeness

**Comment by robinjoseph08** [comment]
LGTM after last test

**Pull request merged**